packages/dd-trace/test/appsec/iast/code_injection.integration.spec.js

```diff
@@ -53,24 +42,79 @@ describe('IAST - code_injection - integration', () => {
   async function testVulnerabilityRepoting (url) {
     await axios.get(url)
 
-    return agent.assertMessageReceived(({ headers, payload }) => {
-      expect(payload[0][0].metrics['_dd.iast.enabled']).to.be.equal(1)
-      expect(payload[0][0].meta).to.have.property('_dd.iast.json')
+    let iastTelemetryReceived = false
+    const checkTelemetry = agent.assertTelemetryReceived(({ headers, payload }) => {
```

Comment:
I don't understand why these tests are working.
The request is finished when we are subscribing to the agent, why are we able to check the logs that happened before? Could this be a source of flakyness?
I think that we should set these asserts before the `axios.get(url)`

Reply:
Ok, I can see more code implemented like this, I think that it is working because there is at least small delay betewen the vuln is detected and reported and the logs are send to the agent.